src/Symfony/Component/Console/Input/ArgvInput.php

```diff
@@ -283,7 +283,7 @@ public function hasParameterOption($values)
                 //   For long options, test for '--option=' at beginning
                 //   For short options, test for '-o' at beginning
                 $leading = 0 === strpos($value, '--') ? $value.'=' : $value;
-                if ($token === $value || 0 === strpos($token, $leading)) {
+                if ($token === $value || !empty($leading) && 0 === strpos($token, $leading)) {
```

Comment:
Let's try to avoid using `empty()` as it catches too many cases, like`0` and others.

Reply:
The reason for this conditional is to avoid values of $leading that cause problems for strpos; therefore, I think that `empty` is entirely appropriate, since passing `0` et. al. to strpos would also be problematic.

I'll go ahead and make this change later today for the sake of style, though.

Reply:
@greg-1-anderson `'0'` is empty as well: https://3v4l.org/vuoiA

Use `'' !== $value` instead

Reply:
@stof The point of my comment is that the purpose of adding a conditional here is only to prevent `strpos` from emitting a warning if the caller of these functions passes a bad value. With `!==`, we protect only against the empty string, and allow the warning to happen if uses pass `0` or `null`.

That said, I will follow the provided advice for the sake of style, although it makes the guard less effective.